src/functions/search-stac-api.ts

```diff
@@ -66,17 +78,27 @@ export default async function searchStacApi(
       dateConstraint = `&datetime=../${endDate}`
     }
 
-    // Create the CQL filter and encode it for URL
-    const cqlFilter = encodeURIComponent(
-      `eo:cloud_cover<${cloudCover} AND s2:mgrs_tile='${mgrsTileId}'`,
-    )
-
     // Construct the URL with query parameters and a limit of 20 results
-    let searchUrl = `https://planetarycomputer.microsoft.com/api/stac/v1/search?collections=sentinel-2-l2a${dateConstraint}&filter-lang=cql2-text&filter=${cqlFilter}&limit=20`
+    let searchUrl = `https://earth-search.aws.element84.com/v1/collections/sentinel-2-l2a/items?${dateConstraint}&limit=20`
+
+    // Add bbox parameter if provided
+    if (bbox && bbox.length === 4) {
+      // Convert from EPSG:3857 to EPSG:4326 (WGS84) if needed
+      const [minX, minY, maxX, maxY] = bbox
+
+      // Import the transform function from OpenLayers
+      const { transform } = await import('ol/proj')
+
+      // Transform coordinates from EPSG:3857 to EPSG:4326
+      const [minLon, minLat] = transform([minX, minY], 'EPSG:3857', 'EPSG:4326')
+      const [maxLon, maxLat] = transform([maxX, maxY], 'EPSG:3857', 'EPSG:4326')
+
+      searchUrl += `&bbox=${minLon},${minLat},${maxLon},${maxLat}`
```

Comment:
Not a big deal, but next time you could write something like this as simple as
```suggestion
      // Convert from EPSG:3857 to EPSG:4326 (WGS84) if needed

      // Import the transformExtent function from OpenLayers
      const { transformExtent } = await import('ol/proj')

      // Transform coordinates from EPSG:3857 to EPSG:4326
      const lonLatBbox = transformExtent(bbox, 'EPSG:3857', 'EPSG:4326')

      searchUrl += `&bbox=${lonLatBbox.join(',')}`
```